cmake/python.cmake

```diff
@@ -9,5 +9,29 @@ if (WIN32)
   set(ENV{PYTHONIOENCODING} "utf-8")
 endif()
 
-find_package(Python3 3.6 REQUIRED)
+set(PYTHON_MINIMUM_REQUIRED 3.6)
+
+# We are using foreach here, instead of find_program(PYTHON_EXECUTABLE_SYSTEM_DEFAULT "python" "python3")
+# cause just using find_program directly could result in a python2.7 as python, and not finding a valid python3.
+foreach(PYTHON_PREFER ${PYTHON_PREFER} "python" "python3")
```

Comment:
I'd suggest that the order should be "python3" "python". AFAIK, it's official recommendation from the Python project that Python3 executable should be named "python3". I know that some e.g. Linux distros don't follow that, and after they think that they migrated off Python2 use "python" to refer to Python3, but that's their own choice (known to lead to good deal of user confusion). I'm sure we should (try) to support such, but still should prioritize the common case.

Reply:
> AFAIK, it's official recommendation from the Python project that Python3 executable should be named "python3"

if you have a link to that recommendation, then it would be great.
I have no personal preference on this, but I would like us to avoid having later discussions on this, and changing it later.

So being able to refer to a recommendation, and follow that, is the optimal solution.

Reply:
> if you have a link to that recommendation, then it would be great.

That appears to be https://www.python.org/dev/peps/pep-0394/ . The complete picture is somewhat more complicated:

> We expect Unix-like software distributions (including systems like macOS and Cygwin) to install the python2 command into the default path whenever a version of the Python 2 interpreter is installed, and the same for python3 and the Python 3 interpreter.

> Distributors may choose to set the behavior of the python command as follows:
> * python2,
> * python3,
> * not provide python command,

But the conclusion would be the same as my original puts: as long as Python3 installed, this PEP stipulates that "python3" command should exists. Whereas it says that "python" may not even exist, and if it does, it may refer to an arbitrary version.

So, recommendation remains the same: while searching for a valid interpreter, try "python3" before "python", it would lead to a quicker success.

Reply:
Thanks for the link.

This is interesting
> If the python command is installed, it is expected to invoke either the same version of Python as the python3 command or as the python2 command.

so if `python` points to a Python3 version, then it is expected to point to same version as `python3`, so this situation is illegal:
```
/usr/bin/python --> python3.6
/usr/bin/python3 --> python3.8
```
and thus it doesn't matter whether we test `python` before `python3` or the other way around.

Reply:
> and thus it doesn't matter whether we test python before python3 or the other way around.

Yes, except on most systems for the foreseeable future "python" points to "python2", so if you start with running "python --version", you likely do useless invocation, and will have to follow with "python3 --version", so makes sense to start with the latter. Anyway, merged already.

Reply:
agree to that.
And it was already merged when I commented.

Will keep in mind and if doing the `west` fix, then order can change.